launcher/net/Download.cpp

```diff
@@ -48,12 +51,106 @@
 #include "BuildConfig.h"
 #include "Application.h"
 
+#include "net/Logging.h"
+#include "net/NetAction.h"
+
 namespace Net {
 
+QString truncateUrlHumanFriendly(QUrl &url, int max_len, bool hard_limit = false)
+{   
+    auto display_options = QUrl::RemoveUserInfo | QUrl::RemoveFragment | QUrl::NormalizePathSegments;
+    auto str_url = url.toDisplayString(display_options);
+    if (str_url.length() <= max_len)
+        return str_url;
+
+    /* this is a PCRE regular expression that splits a URL (given by the display rules above) into 5 capture groups
+     * the scheme (ie https://) is group 1
+     * the host (with trailing /) is group 2
+     * the first part of the path (with trailing /) is group 3
+     * the last part of the path (with leading /) is group 5
+     * the remainder of the URL is in the .* and in group 4
+     *
+     * See: https://regex101.com/r/inHkek/1
+     * for an interactive breakdown
+     */ 
+    QRegularExpression re(R"(^([\w]+:\/\/)([\w._-]+\/)([\w._-]+\/)(.*)(\/[^]+[^]+)$)");
```

Comment:
We could also extract some of these using QUrl's own parser. And as URLs can be very complex, I would prefer if we relied on a proper parser than using RegEx.

Reply:
I actually tried using qurl first but I ended up needing most of this regular expression anyway to split the path. Adding a portion to capture the scheme and host simplifyed the code significantly. 

The point isn't to parse the parts of a url but to place an elipsus in the middle of the path intelagetly to truncate it's display. For that purpose I feel this approach with the regular expression is quite suitable seeing as there is no need to properly parse port or user athority or query fragments. (these are explicitly stripped out beforehand)

Reply:
That said I think I came up with a non regex version.